spec/inputs/couchdb_changes_spec.rb

```diff
@@ -1 +1,507 @@
+
+    before do
+      buildup
+      createuser
+    end
+
+    ftw = FTW::Agent.new
+
+    config <<-CONFIG
+    input {
+      couchdb_changes {
+        db => "db"
+        host => "127.0.0.1"
+        timeout => 2000
+        always_reconnect => false
+        sequence_path => "#{sequence}"
+        type => "couchdb"
+        username => "#{user}"
+        password => "#{pass}"
+      }
+    }
+    output {
+      elasticsearch {
+        action => "%{[@metadata][action]}"
+        document_id => "%{[@metadata][_id]}"
+        host => "127.0.0.1"
+        index => "#{index}"
+        protocol => "http"
+      }
+    }
+    CONFIG
+      
+    agent do
+      # Verify the count
+      ftw.post!("http://127.0.0.1:9200/#{index}/_refresh")
+      data = ""
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_count?q=*")
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      count = result["count"]
+      insist { count } == 10
+      # Get the docs and do a couple spot checks
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_search?q=*&size=10")
+      data = ""
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      result["hits"]["hits"].each do |doc|
+        case doc["_id"]
+        when 3
+          insist { doc["_source"]["name"] } == "Captain America"
+        end
+      end
+    end
+    
+    after do
+      deleteuser
+      teardown
+    end
+  end
+
+  describe "Test Secure Connection", :elasticsearch => true, :couchdb => true do
+    include Helpers
+    sequence = "/tmp/.couchdb_seq"
+    index = "couchdb_test"
+    ca_file = File.dirname(__FILE__) + "/ca_cert.pem"
+
+    before do
+      buildup
+    end
+
+    ftw = FTW::Agent.new
+
+    config <<-CONFIG
+    input {
+      couchdb_changes {
+        db => "db"
+        host => "localhost"
+        port => 6984
+        timeout => 2000
+        always_reconnect => false
+        sequence_path => "#{sequence}"
+        type => "couchdb"
+        secure => true
+        ca_file => "#{ca_file}"
+      }
+    }
+    output {
+      elasticsearch {
+        action => "%{[@metadata][action]}"
+        document_id => "%{[@metadata][_id]}"
+        host => "127.0.0.1"
+        index => "#{index}"
+        protocol => "http"
+      }
+    }
+    CONFIG
+      
+    agent do
+      # Verify the count
+      ftw.post!("http://127.0.0.1:9200/#{index}/_refresh")
+      data = ""
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_count?q=*")
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      count = result["count"]
+      insist { count } == 10
+      # Get the docs and do a couple spot checks
+      response = ftw.get!("http://127.0.0.1:9200/#{index}/_search?q=*&size=10")
+      data = ""
+      response.read_body { |chunk| data << chunk }
+      result = LogStash::Json.load(data)
+      result["hits"]["hits"].each do |doc|
```

Comment:
similar here to previous comment; this will pass if doc id 8 is not in the results.